src/storage/observer_worker.rs

```diff
@@ -51,6 +58,57 @@ impl ObserverWorker {
         self.try_update().await
     }
 
+    async fn predicate_wrapper(&self, predicate: &Option<ActiveBlobPred>) -> bool {
+        if let Some(predicate) = predicate {
+            predicate(self.inner.active_blob_stat().await)
+        } else {
+            true
+        }
+    }
+
+    async fn close_active_blob(&self, msg: Msg) -> Result<()> {
+        if self.predicate_wrapper(&msg.predicate).await {
+            let _lock = self.async_oplock.lock().await;
+            if self.predicate_wrapper(&msg.predicate).await {
+                self.inner.close_active_blob().await?;

```

Comment:
Lock logic is duplicated in all 4 msg processors, and content of the message is not used for any specific work. Probably it's better to move this logic into separate func (or into tick), and use `optype` inside this func to select `inner` action?

Reply:
I moved it to tick and remembered, what I didn't like in this solution: now this function looks like it's written in LISP :D
It's not a problem for me, but firstly it looked weird)

Reply:
Resolve conversation if now it looks as you expected please